client/packages/common/src/ui/forms/JsonForms/components/Array.tsx

```diff
@@ -76,6 +78,35 @@ const findIndexOfRemoved = (base: string[], newList: string[]): number => {
   return base.length - 1;
 };
 
+const searchRanking = {
+  STARTS_WITH: 2,
+  CONTAINS: 1,
+  NO_MATCH: 0,
+} as const;
+
+const filterOptions = (
+  options: string[],
+  { inputValue }: { inputValue: string }
+) => {
+  const searchTerm = inputValue.toLowerCase();
+  const filteredOptions = options
+    .map(option => {
+      const lowerCaseOption = option.toLowerCase();
+
+      const rank = lowerCaseOption.startsWith(searchTerm)
+        ? searchRanking.STARTS_WITH
+        : lowerCaseOption.includes(searchTerm)
+        ? searchRanking.CONTAINS
+        : searchRanking.NO_MATCH;
+      return { option, rank };
+    })
+    .filter(({ rank }) => rank !== searchRanking.NO_MATCH)
+    .sort((a, b) => b.rank - a.rank)
```

Comment:
Wouldn't it be better to also sort with in the STARTS_WITH and CONTAINS groups? e.g.

```typescript
.sort((a, b) => {
  if (b.rank != a.rank) {
    return b.rank - a.rank;
  }
  return b.localeCompare(a);
})
```

Otherwise the search results could theoretically become unstable (?). 

However, it works very well so think its good to leave as it is and I guess the original order in the list is being preserved?

Reply:
Note, I don't thing that performance is an issue here and the code could become a bit more readable if you split the filter and sort part, i.e. first filter the list by `includes` then sort. However, your version is easy to understand as well.

Reply:
> Wouldn't it be better to also sort with in the STARTS_WITH and CONTAINS groups? e.g.
> 
> ```ts
> .sort((a, b) => {
>   if (b.rank != a.rank) {
>     return b.rank - a.rank;
>   }
>   return b.localeCompare(a);
> })
> ```
> 
> Otherwise the search results could theoretically become unstable (?).
> 
> However, it works very well so think its good to leave as it is and I guess the original order in the list is being preserved?

I thought about that too, but wasn't sure. Can add them in (: